lib/fluent/plugin/out_s3.rb

```diff
@@ -122,6 +123,8 @@ def desc(description)
     config_param :signature_version, :string, :default => nil # use nil to follow SDK default configuration
     desc "Given a threshold to treat events as delay, output warning logs if delayed events were put into s3"
     config_param :warn_for_delay, :time, :default => nil
+    desc "Custom temp dir for compressor, instead of system temp directory."
+    config_param :customer_tmp_dir, :string, :default => Dir.tmpdir
```

Comment:
And passing `nil` to second argument is better beacuse it keeps exsiting behaviour.
Of course, Tempfile uses tmpdir internally but code becomes small.
We can remove `require 'tmpdir'` from plugin code.